core/src/main/java/com/netflix/conductor/model/WorkflowModel.java

```diff
@@ -101,11 +101,22 @@ public boolean isSuccessful() {
     // Capture the failed taskId if the workflow execution failed because of task failure
     private String failedTaskId;
 
+    private Status previousStatus;
+
+    public Status getPreviousStatus() {
+        return previousStatus;
+    }
+
+    public void setPreviousStatus(Status status) {
+        this.previousStatus = status;
+    }
+
     public Status getStatus() {
         return status;
     }
 
     public void setStatus(Status status) {
+        setPreviousStatus(this.status);
```

Comment:
Should we worry about `previousStatus` getting overridden if `setStatus` is called twice?

Reply:
Good catch! Added a check for that